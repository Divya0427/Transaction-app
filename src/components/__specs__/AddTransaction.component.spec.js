import React from 'react';
import { render } from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AddTransactionComponent from '../AddTransaction.Component';

Enzyme.configure({ adapter: new Adapter() });

describe('AddTransactionComponent Component', () => {
  let enzymeWrapper;
  let props;

  beforeEach(() => {
    props = {
      actions: {
        addTransactionItem: jest.fn()
      },

    };

    enzymeWrapper = shallow(<AddTransactionComponent {...props} />);
  });

  it('should render without crashing', () => {
    const div = document.createElement('div');
    render(<AddTransactionComponent {...props} />, div);
  });

  it('should render the component', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should contain a form element', () => {
    expect(enzymeWrapper.find('form').length).toEqual(1);
  });

  it('should contain an textarea field', () => {
    expect(enzymeWrapper.find('textarea[name="tweetBody"]').length).toEqual(1);
  });

  it('should update the state onChange event on the textarea field', () => {
    enzymeWrapper.find('textarea[name="tweetBody"]').props().onChange({
      target: {
        name: 'tweetBody',
        value: 'Test Tweet',
      },
    });
    expect(enzymeWrapper.state('tweetBody')).toBeDefined();
    expect(enzymeWrapper.state('tweetBody')).toEqual('Test Tweet');
    expect(enzymeWrapper).toMatchSnapshot();
  });


  describe('when the tweetBody in state has a value', () => {
    it('should call addTransactionItem on form submit', () => {
      const preventDefault = jest.fn();

      enzymeWrapper.find('form').simulate('submit', { preventDefault });
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });

  describe('when the tweetBody in state has a value', () => {
    const now = Date.now();
    Date.now = jest.fn;

    it('should call addTransactionItem on form submit', () => {
      const preventDefault = jest.fn();
      enzymeWrapper.setState({ tweetBody: 'Test Tweet', date: Date.now() });
      expect(enzymeWrapper.instance().props.actions.addTransactionItem.mock.calls.length).toBe(0);
      enzymeWrapper.find('form').simulate('submit', { preventDefault });
      expect(enzymeWrapper.instance().props.actions.addTransactionItem.mock.calls.length).toBe(1);
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
