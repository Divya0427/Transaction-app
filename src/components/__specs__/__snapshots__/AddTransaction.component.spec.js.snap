// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTransactionComponent Component should render the component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTransactionComponent
    actions={
        Object {
            "addTransactionItem": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={undefined}
>
        <input
                name="description"
                onChange={undefined}
                placeholder="Description.."
                type="text"
                value={undefined}
        />
        <input
                name="date"
                onChange={undefined}
                type="date"
                value={undefined}
        />
        <input
                min="0"
                name="amount"
                onChange={undefined}
                placeholder="Amount.."
                type="number"
                value={undefined}
        />
        <button
                type="submit"
        >
                Add
        </button>
</form>,
      "id": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="description"
            onChange={undefined}
            placeholder="Description.."
            type="text"
            value={undefined}
/>,
          <input
            name="date"
            onChange={undefined}
            type="date"
            value={undefined}
/>,
          <input
            min="0"
            name="amount"
            onChange={undefined}
            placeholder="Amount.."
            type="number"
            value={undefined}
/>,
          <button
            type="submit"
>
            Add
</button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "description",
            "onChange": undefined,
            "placeholder": "Description..",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "date",
            "onChange": undefined,
            "type": "date",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "min": "0",
            "name": "amount",
            "onChange": undefined,
            "placeholder": "Amount..",
            "type": "number",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={undefined}
>
          <input
                    name="description"
                    onChange={undefined}
                    placeholder="Description.."
                    type="text"
                    value={undefined}
          />
          <input
                    name="date"
                    onChange={undefined}
                    type="date"
                    value={undefined}
          />
          <input
                    min="0"
                    name="amount"
                    onChange={undefined}
                    placeholder="Amount.."
                    type="number"
                    value={undefined}
          />
          <button
                    type="submit"
          >
                    Add
          </button>
</form>,
        "id": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="description"
              onChange={undefined}
              placeholder="Description.."
              type="text"
              value={undefined}
/>,
            <input
              name="date"
              onChange={undefined}
              type="date"
              value={undefined}
/>,
            <input
              min="0"
              name="amount"
              onChange={undefined}
              placeholder="Amount.."
              type="number"
              value={undefined}
/>,
            <button
              type="submit"
>
              Add
</button>,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "description",
              "onChange": undefined,
              "placeholder": "Description..",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "date",
              "onChange": undefined,
              "type": "date",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "min": "0",
              "name": "amount",
              "onChange": undefined,
              "placeholder": "Amount..",
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AddTransactionComponent Component when the tweetBody in state has a value should call addTransactionItem on form submit 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddTransactionComponent
    actions={
        Object {
            "addTransactionItem": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={undefined}
>
        <input
                name="description"
                onChange={undefined}
                placeholder="Description.."
                type="text"
                value={undefined}
        />
        <input
                name="date"
                onChange={undefined}
                type="date"
                value={undefined}
        />
        <input
                min="0"
                name="amount"
                onChange={undefined}
                placeholder="Amount.."
                type="number"
                value={undefined}
        />
        <button
                type="submit"
        >
                Add
        </button>
</form>,
      "id": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="description"
            onChange={undefined}
            placeholder="Description.."
            type="text"
            value={undefined}
/>,
          <input
            name="date"
            onChange={undefined}
            type="date"
            value={undefined}
/>,
          <input
            min="0"
            name="amount"
            onChange={undefined}
            placeholder="Amount.."
            type="number"
            value={undefined}
/>,
          <button
            type="submit"
>
            Add
</button>,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "description",
            "onChange": undefined,
            "placeholder": "Description..",
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "date",
            "onChange": undefined,
            "type": "date",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "min": "0",
            "name": "amount",
            "onChange": undefined,
            "placeholder": "Amount..",
            "type": "number",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={undefined}
>
          <input
                    name="description"
                    onChange={undefined}
                    placeholder="Description.."
                    type="text"
                    value={undefined}
          />
          <input
                    name="date"
                    onChange={undefined}
                    type="date"
                    value={undefined}
          />
          <input
                    min="0"
                    name="amount"
                    onChange={undefined}
                    placeholder="Amount.."
                    type="number"
                    value={undefined}
          />
          <button
                    type="submit"
          >
                    Add
          </button>
</form>,
        "id": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="description"
              onChange={undefined}
              placeholder="Description.."
              type="text"
              value={undefined}
/>,
            <input
              name="date"
              onChange={undefined}
              type="date"
              value={undefined}
/>,
            <input
              min="0"
              name="amount"
              onChange={undefined}
              placeholder="Amount.."
              type="number"
              value={undefined}
/>,
            <button
              type="submit"
>
              Add
</button>,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "description",
              "onChange": undefined,
              "placeholder": "Description..",
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "date",
              "onChange": undefined,
              "type": "date",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "min": "0",
              "name": "amount",
              "onChange": undefined,
              "placeholder": "Amount..",
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Add",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
